tests/osn-tests/src/test_osn_filter.ts

```diff
@@ -24,17 +25,31 @@ describe('osn-filter', () => {
     });
 
     context('# Create', () => {
-        let filter: IFilter
-
         it('should create filter', () => {
+            let filter: IFilter
+
             try {
                 filter = osn.FilterFactory.create('test_filter', 'filter1');
             } catch(e) {
-                throw new Error("failed to create filter");
+                throw new Error(getCppErrorMsg(e));
             }
             
             expect(filter.id).to.equal('test_filter');
             expect(filter.name).to.equal('filter1');
         });
     });
+
+    context('# Types', () => {
+        it('should get all filter types', () => {
+            let filterTypes: string[];
+
+            try {
+                filterTypes = osn.FilterFactory.types();
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
+            }
+
+            expect(filterTypes.length).to.not.equal(0);
```

Comment:
I would test more things here. Only making sure the array is not empty doesn't tell us if the function is correctly working or not.

Reply:
I tried to add and expect to compare the content of the array with the content of another array I initialized but the difference in the filters available on each machine makes this impossible to work. For example, in my machine I have ndi_filter, ndi_audiofilter and permultiplied_alpha_filter and in the machine that the AppVeyor runs these 3 filters are missing. Any other ideas as to what I should test here?

Reply:
What we should do is make sure the array we fetch contains a specific set of values.

Reply:
Figured out what the basic OBS filter types are and now I verify if the array has them